pkgs/stdenv/generic/setup.sh

```diff
@@ -602,9 +606,11 @@ fi
 
 PATH="${_PATH-}${_PATH:+${PATH:+:}}$PATH"
 HOST_PATH="${_HOST_PATH-}${_HOST_PATH:+${HOST_PATH:+:}}$HOST_PATH"
+export XDG_DATA_DIRS="${_XDG_DATA_DIRS-}${_XDG_DATA_DIRS:+${XDG_DATA_DIRS:+:}}${XDG_DATA_DIRS-}"
```

Comment:
Is this going to affect wrapGAppsHook or wrapQtAppsHook in any way?

Reply:
It shouldn't affect those.
This variable is set in the build environment, not in the wrapper scripts.

Reply:
cc @NixOS/freedesktop

Reply:
Oh, so being set in the build environment, but during fixup it shouldn't be?

Reply:
Well it's still set during `fixupPhase` but it doesn't affect the wrapper writing functions. Although, it can affect call sites if they are like `wrapProgram --set XDG_DATA_DIRS $XDG_DATA_DIRS`, so expanding `$XDG_DATA_DIRS` before invoking `wrapProgram`, but I don't think we have that?

Reply:
Yeah, `wrapGAppsHook` uses different environment variables for populating `XDG_DATA_DIRS` for wrapper. So unless one of those variables is populated from `XDG_DATA_DIRS` (do not see any such thing when `grep`ing `sh` files for `XDG_DATA_DIRS` so it would have to be indirectly through some program).

The only possible issue that comes to mind is that it further distances the build environment from the runtime one, possibly obscuring some missing dependencies until runtime (i.e. `(install)CheckPhase` passes but program will crash at runtime). But perhaps the difference is huge enough already so this might not be so much worse in proportion.